### ISSUE TITLE
junying/event_filter:  query delegations   panic

### ISSUE THREAD

**Issue by youngqqcn** [orientwalt/htdf#40]
```
$ hscli query staking delegations htdf1pq0xeagjvd8rdwjgmg66jnpx53gug45sr8xu2g
panic: Bytes left over in UnmarshalBinaryLengthPrefixed, should read 10 more bytes but have 122

goroutine 1 [running]:
github.com/tendermint/go-amino.(*Codec).MustUnmarshalBinaryLengthPrefixed(...)
        /home/yqq/go/pkg/mod/github.com/tendermint/go-amino@v0.16.0/amino.go:316
github.com/orientwalt/htdf/client/context.CLIContext.QuerySubspace(0xc000225ce0, 0x0, 0x1588e20, 0xc0012433e0, 0xc001033e90, 0x9, 0x0, 0x0, 0x1555800, 0xc000138008, ...)
        /home/yqq/work/htdf/client/context/query.go:64 +0x225
github.com/orientwalt/htdf/x/staking/client/cli.GetCmdQueryDelegations.func1(0xc00106e840, 0xc00124c660, 0x1, 0x1, 0x0, 0x0)
        /home/yqq/work/htdf/x/staking/client/cli/query.go:209 +0x190
github.com/spf13/cobra.(*Command).execute(0xc00106e840, 0xc00124c640, 0x1, 0x1, 0xc00106e840, 0xc00124c640)
        /home/yqq/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:850 +0x47c
github.com/spf13/cobra.(*Command).ExecuteC(0xc000242580, 0xcc0d85, 0xc000242580, 0x12959ff)
        /home/yqq/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        /home/yqq/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:895
github.com/tendermint/tendermint/libs/cli.Executor.Execute(0xc000242580, 0x14239e8, 0x2, 0xc0010321a0)
        /home/yqq/go/pkg/mod/github.com/orientwalt/tendermint@v0.99.2/libs/cli/setup.go:89 +0x3c
main.main()
        /home/yqq/work/htdf/cmd/hscli/main.go:118 +0x7c5
```

**Comment by youngqqcn** [comment]
`delegation` subcommand  is work

```
hscli query staking delegation htdf1r0lpw30p7meu2hlw2qxxr86a5qf74r83plmvl5  htdfvaloper1r0lpw30p7meu2hlw2qxxr86a5qf74r83txr4gl
```

**Comment by youngqqcn** [comment]
API `localhost:1317/distribution/delegators/{delegator_addr}/rewards/{validator}`  always return `null`

```
curl  localhost:1317/distribution/delegators/htdf12tmus4ke89zlppejdrz60q34adqzx2exk58rup/rewards/htdfvaloper12tmus4ke89zlppejdrz60q34adqzx2exudl6t2
null
```

**Comment by bluesky0125** [comment]
this issue resolved since commit: bc17054f9c71124fc6a59cd1c0c84dc0460cc21d

**Comment by bluesky0125** [comment]
this issue resolved since commit: bc17054f9c71124fc6a59cd1c0c84dc0460cc21d
